include/odp/api/spec/schedule.h

```diff
@@ -368,6 +368,27 @@ void odp_schedule_order_lock(unsigned lock_index);
 void odp_schedule_order_unlock(unsigned lock_index);
 
 /**
+ * Release existing ordered context lock and acquire a new lock
+ *
+ * This call is valid only when holding an ordered synchronization context.
+ * Release a previously locked ordered context lock and acquire a new ordered
+ * context lock. This call is valid only when there is a single ordered context
+ * lock active. Results are undefined if this call is made when multiple ordered
+ * context locks are acquired in nested fashion.
+ * This call does not release the existing ordered context and only switches
+ * from one ordered lock to another ordered lock within the same ordered
+ * context.
+ *
+ * @param lock_index	Index of the ordered lock in the current context to be
+ *			acquired. Previously acquired ordered lock is released.
+ *			Must be in the range 0...odp_queue_lock_count() - 1.
+ *
+ * @see odp_schedule_order_lock(), odp_schedule_order_unlock()
+ *
+ */
+void odp_schedule_order_unlock_lock(uint32_t lock_index);
```

Comment:
To match other function prototypes, should be **unsigned** instead of uint32_t.

Reply:
+*  This call is valid only when there is a single ordered context
+ * lock active. Results are undefined if this call is made when multiple ordered
+ * context locks are acquired in nested fashion.

Why nested locks are banned? Thinks it should not make difference. Unlock_lock(C) is equivalent to unlock(B) + lock(C) under.

lock(A)
lock(B)
unlock(B)
lock(C)
...

Reply:
"This call does not release the existing ordered context and only switches
from one ordered lock to another ordered lock within the same ordered
context."

I'd remove this since this call is not different to lock/unlock calls in this regard

Reply:
Was there some benefit of not having index of the unlocked lock as a parameter ? Can't remember if I suggested it myself.  Anyway, implementation would simplify and it would still be as robust as unlock() (which application may pass bad lock index).